lib/Serialization/SerializeSIL.cpp

```diff
@@ -2469,6 +2491,54 @@ writeSILDefaultWitnessTable(const SILDefaultWitnessTable &wt) {
   }
 }
 
+// SWIFT_ENABLE_TENSORFLOW
+void SILSerializer::
+writeSILDifferentiabilityWitness(const SILDifferentiabilityWitness &dw) {
+  Mangle::ASTMangler mangler;
+  auto mangledKey = mangler.mangleSILDifferentiabilityWitnessKey(dw.getKey());
+  size_t nameLength = mangledKey.size();
+  char *stringStorage =
+      static_cast<char *>(StringTable.Allocate(nameLength, 1));
+  std::memcpy(stringStorage, mangledKey.data(), nameLength);
```

Comment:
Use `std::uninitialized_copy` instead?

Reply:
I get some errors using `std::uninitialized_copy`:
```
/usr/bin/../lib/gcc/x86_64-linux-gnu/7.4.0/../../../../include/c++/7.4.0/bits/stl_construct.h:75:13: error: static_cast from 'const char *' to 'void *' is not allowed
    { ::new(static_cast<void*>(__p)) _T1(std::forward<_Args>(__args)...); }
            ^~~~~~~~~~~~~~~~~~~~~~~
/usr/bin/../lib/gcc/x86_64-linux-gnu/7.4.0/../../../../include/c++/7.4.0/bits/stl_uninitialized.h:83:8: note: in instantiation of function template specialization 'std::_Construct<const char, char &>' requested here
                std::_Construct(std::__addressof(*__cur), *__first);
                     ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/7.4.0/../../../../include/c++/7.4.0/bits/stl_uninitialized.h:134:2: note: in instantiation of function template specialization 'std::__uninitialized_copy<false>::__uninit_copy<char *, const char *>' requested here
        __uninit_copy(__first, __last, __result);
        ^
/home/danielzheng/swift-merge/swift/lib/Serialization/SerializeSIL.cpp:2503:8: note: in instantiation of function template specialization 'std::uninitialized_copy<char *, const char *>' requested here
  std::uninitialized_copy(stringStorage, stringStorage + nameLength,
       ^
1 error generated.
```

I copied this logic including `std::memcpy` from existing code. I'd prefer to defer this minor improvement to speed up progress, existing code can also be updated.